plugins/SegmentEditor/Model.php

```diff
@@ -129,6 +129,75 @@ public function getSegmentByDefinition($definition)
         return $segment;
     }
 
+    /**
+     * Gets a list of segments that have been deleted in the last week and therefore may have orphaned archives.
+     * @param Date $date Segments deleted on or after this date will be returned.
+     * @return array of segments. The segments are only populated with the fields needed for archive invalidation
+     * (e.g. definition, enable_only_idsite).
+     * @throws \Exception
+     */
+    public function getSegmentsDeletedSince(Date $date)
+    {
+        $dateStr = $date->getDatetime();
+        $sql = "SELECT DISTINCT definition, enable_only_idsite FROM " . Common::prefixTable('segment')
+            . " WHERE deleted = 1 AND ts_last_edit >= ?";
+        $deletedSegments = Db::fetchAll($sql, array($dateStr));
+
+        if (empty($deletedSegments)) {
+            return array();
+        }
+
+        $existingSegments = self::getExistingSegmnetsLike($deletedSegments);
+
+        foreach ($deletedSegments as $i => $deleted) {
+            $deletedSegments[$i]['idsites_to_preserve'] = array();
+            foreach ($existingSegments as $existing) {
+                if ($existing['definition'] != $deleted['definition']) {
+                    continue;
+                }
+
+                if (
+                    $existing['enable_only_idsite'] == $deleted['enable_only_idsite']
+                    || $existing['enable_only_idsite'] == 0
+                ) {
+                    // There is an identical segment (for either the specific site or for all sites) that is active
+                    // The archives for this segment will therefore still be needed
+                    unset($deletedSegments[$i]);
+                    break;
+                } elseif ($deleted['enable_only_idsite'] == 0) {
+                    // It is an all-sites segment that got deleted, but there is a single-site segment that is active
+                    // Need to make sure we don't erase the segment's archives for that particular site
+                    $deletedSegments[$i]['idsites_to_preserve'][] = $existing['enable_only_idsite'];
+                }
+            }
+        }
+
+        return $deletedSegments;
+    }
+
+    private static function getExistingSegmnetsLike(array $segments)
+    {
+        $whereClauses = array();
```

Comment:
be good to add a little check `if (empty($segments)) { return array(); }`. I know this is done in the method that is calling this method already, be just good to add this here as well in case the method will be used in the future somewhere else too and then it prevents the case of having a broken sql..   
also be good to remove the `static` and make it a regular method